mono/mini/driver.c

```diff
@@ -1920,18 +1920,26 @@ mono_main (int argc, char* argv[])
 		} else if (strcmp (argv [i], "--nollvm") == 0){
 			mono_use_llvm = FALSE;
 		} else if ((strcmp (argv [i], "--interpreter") == 0) || !strcmp (argv [i], "--interp")) {
-#ifdef ENABLE_INTERPRETER
+#ifndef DISABLE_INTERPRETER
 			mono_use_interpreter = TRUE;
 #else
 			fprintf (stderr, "Mono Warning: --interpreter not enabled in this runtime.\n");
 #endif
+#if !defined(TARGET_AMD64) && !defined(TARGET_ARM) && !defined(TARGET_ARM64)
```

Comment:
We do want `HOST_*` as the target is only useful for the cross compiler. Or is it the case that we do not have these defines and we just use `TARGET_*`?

Reply:
yeah, `HOST_*`  is only available for platforms, but not architectures. I added a check for `MONO_CROSS_COMPILE` instead and make the runtime bail out on that, because I can't think of a use case where you want to use the interpreter with a cross-compiling runtime...